test/testcanvas.go

```diff
@@ -167,6 +168,32 @@ func (c *testCanvas) Capture() image.Image {
 	return img
 }
 
+// CanvasItem is a part of a canvas and returned by InspectCanvasItems().
+type CanvasItem struct {
+	Type   string
+	Object fyne.CanvasObject
+}
+
+// InspectCanvasItems returns all CanvasItems starting at the given CanvasObject which is laid out with the given size.
+func InspectCanvasItems(o fyne.CanvasObject, size fyne.Size) (objects []CanvasItem) {
```

Comment:
This name is quite abstract, really what this is doing is simply flattening a tree of objects?
Every usage of this method passes an item and it's size - could it be simplified to not pass the size at all?
I wonder if "ListCanvasObjects(o fyne.CanvasObject) []fyne.CanvasObject" or "ListObjectTree(o fyne.CanvasObject) []fyne.CanvasObject" is all that's needed?

In fact, seeing that in every usage you are then iterating over them perhaps a foreach pattern instead?
"ForEachCanvasObject(o fyne.CanvasObject, func(c fyne.CanvasObject))"? in essence it's a simpler version of the internal walk method we use.